gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapperTest.java

```diff
@@ -230,7 +227,82 @@ public void When_MappingMedicationRequestWithAvailabilityTimeOnlyInEhrExtract_Ex
+        var ehrExtract = unmarshallEhrExtract("ehrExtract_hasAuthorTime.xml");
+        var expectedAuthoredOn = DateFormatUtil.parseToDateTimeType("20100115");
+
+        when(medicationRequestPlanMapper.mapToPlanMedicationRequest(any(), any(), any(), any())).thenReturn(new MedicationRequest());
+        when(medicationRequestOrderMapper.mapToOrderMedicationRequest(any(), any(), any(), any())).thenReturn(new MedicationRequest());
+        when(medicationStatementMapper.mapToMedicationStatement(any(), any(), any(), any(), any())).thenReturn(new MedicationStatement());
+        when(medicationMapper.createMedication(any())).thenReturn(new Medication());
+
+        var resources = medicationRequestMapper.mapResources(ehrExtract, (Patient) new Patient().setId(PATIENT_ID), List.of(),
+                PRACTISE_CODE);
+
+        assertThat(resources.size()).isEqualTo(EXPECTED_RESOURCES_MAPPED);
+
+        var medicationRequest = resources
+                .stream()
+                .filter(resource -> ResourceType.MedicationRequest.equals(resource.getResourceType()))
+                .map(MedicationRequest.class::cast)
+                .findFirst()
+                .get();
+
+        assertThat(medicationRequest.getAuthoredOnElement().getValue()).isEqualTo(expectedAuthoredOn.getValue());
+    }
+
+    @Test
+    public void When_MappingMedicationRequestWithAuthoredOnValidDateInExtractAndComposition_Expect_AuthoredOnToUseCompositionValue() {
+        var ehrExtract = unmarshallEhrExtract("ehrExtract_hasAuthorTimeInExtract.xml");
+        var expectedAuthoredOn = DateFormatUtil.parseToDateTimeType("20100115");
+
+        when(medicationRequestPlanMapper.mapToPlanMedicationRequest(any(), any(), any(), any())).thenReturn(new MedicationRequest());
+        when(medicationRequestOrderMapper.mapToOrderMedicationRequest(any(), any(), any(), any())).thenReturn(new MedicationRequest());
+        when(medicationStatementMapper.mapToMedicationStatement(any(), any(), any(), any(), any())).thenReturn(new MedicationStatement());
+        when(medicationMapper.createMedication(any())).thenReturn(new Medication());
+
+        var resources = medicationRequestMapper.mapResources(ehrExtract, (Patient) new Patient().setId(PATIENT_ID), List.of(),
+                PRACTISE_CODE);
+
+        assertThat(resources.size()).isEqualTo(EXPECTED_RESOURCES_MAPPED);
+
+        var medicationRequest = resources
+                .stream()
+                .filter(resource -> ResourceType.MedicationRequest.equals(resource.getResourceType()))
+                .map(MedicationRequest.class::cast)
+                .findFirst()
+                .get();
+
+        assertThat(medicationRequest.getAuthoredOnElement().getValue()).isEqualTo(expectedAuthoredOn.getValue());
+    }
+
+    @Test
+    public void When_MappingMedicationRequestWithAuthoredOnValidDateInExtractOnly_Expect_AuthoredOnToUseExtractValue() {
```

Comment:
I'm a bit confused by this test as the title of the test says "InExtractOnly", but the test appears to be using a date in the `MedicationStatement` element.